lib/python/rose/apps/rose_ana.py

```diff
@@ -360,6 +418,9 @@ def _load_analysis_modules(self):
                 try:
                     self.modules.add(__import__(module_name))
                 except Exception as err:
+                    # Note: We intentionally don't re-raise the exception
+                    # here, as we want to avoid a single mistake in a user
+                    # supplied method bringing down the entire task
```

Comment:
Should this:
* catch only `ImportError`?
* report a warning instead of a fail?

Reply:
Hmm... I suppose I could use ``ImportError`` although I figured since using ``__import__`` felt a little bit non-standard it would be safer to catch anything - do you think it can be relied upon to ``raise`` the right exception type?

I can change the prefix to ``[WARN]`` but should that be ``KIND_ERR`` or ``KIND_OUT`` (as there isn't a ``KIND_WARN``)?

Reply:
`import` uses `__import__` internally, according to the documentation of the Python's built-in functions, so it should be sufficient to catch only `ImportError`. You may want to catch `TypeError` as well (to catch non-string argument) or you can explicitly convert the argument to a `str` before passing it into the function.

Yes, you can easily change that by switching to `KIND_ERR` which will use the warning prefix by default.